app/components/join-osf-banner/component.ts

```diff
@@ -0,0 +1,26 @@
+import { action, computed } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import { SessionService } from 'ember-simple-auth/addon/services/session';
+import Analytics from '../../services/analytics';
+
+export default class JoinOsfBanner extends Component.extend() {
+    @service analytics: Analytics;
+    @service session: SessionService;
```

Comment:
Since the session service is not TypeScript, this doesn't do anything useful. Instead, we should create a declarations file that types the service (or, at least, the parts we use).

Reply:
So that's not happening with the other typescript tickets, then? Okay, I'll do some research.